### PR TITLE
Proactively suggest an independent second-harness review flow

### PR DESCRIPTION
**Pull request by realtonyyoung**
Linear: **AI-2175**

## What & why

Most vibe coders never run an independent code review — not with a second coding agent, nor outside the implementing session. Review flows already exist and the reviewer can be any installed harness independent of the driver, but the `review-flows` skill is strictly opt-in and never proposes itself, so beginners never discover it.

This adds a **model-driven skill that proactively offers an independent review flow** at the two natural milestones — right after a spec/design is finalized, and right after implementation is complete — and recommends a reviewer harness that will *actually run* for this repo.

## Changes (purely additive — one new skill + one new tool)

- **`list_reviewer_vendors`** MCP tool in `kcap mcp flows` (agent-only, read-only). Reads `GET /api/daemons` and computes, client-side, the reviewer vendors that can run an unattended review for **this** repo now — the intersection of same-machine daemons hosting the repo × their unattended vendors. Every empty result is disambiguated by a single typed `reason` (repo_unresolved · schema_skew · lookup_failed · no_daemons_connected · no_repo_hosting_daemon · no_unattended_reviewer). Conservative `model_override` (AND across daemons). Tolerant daemon parsing (case-insensitive; malformed records skipped + counted; all-unparseable ⇒ schema_skew).
- **`suggest-review-flow`** skill — offers at spec-complete → `spec-review` and implementation-complete → `code-review`; prefers a reviewer ≠ the driver, falls back to a same-vendor independent flow, and maps an empty result to one-line unlock guidance. **Never auto-runs** — it offers, and hands off to `review-flows` only on explicit consent. Ships identically to all nine harnesses; the driver vendor is **inferred at call time** from harness env (not stamped, since six harnesses share `~/.agents/skills`), echoed as `driver_vendor`, unknown ⇒ null ⇒ no "different model" claim.
- **`DriverVendor`** — env-marker inference (claude, codex today; ambiguous/unverified ⇒ null).
- Registered in `AgentsSkillsInstaller.SourceNames` + `help-plugin.txt` + `README.md` (all test-pinned); conformance pins on the triggers and safety guardrails.
- Docs: `docs/eval/suggest-review-flow.md` — the reproducible triggering-eval corpus + methodology.

## No server change

Verified against kcap-server: `GET /api/daemons` already returns `DaemonInfo` with `repoPaths` + `unattendedVendors` + `machineId` + `unattendedVendorCapabilities`, so this is CLI-only.

## Review

The design (AI-2175) was hardened over a **Codex spec-review flow, 6 rounds to clean** (repo-affinity, a circular eval gate, and a branch-name identity collision were all caught there). Deterministic unit/contract tests cover the tool's aggregation, every typed reason + precedence, partial-record handling, and identity edge cases; static conformance pins carry the skill's model-driven surface. AOT publish is clean.

No submodule pin bump is included (that's the maintainer's).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/Capacitor.Cli/Commands/McpFlowsServer.cs [hunk 1]
```diff
@@ -380,6 +384,30 @@ internal static async Task<string> HandleToolCallAsync(
...
+                    client, apiRoot, (c, ct) => c.GetAsync(apiRoot + "/api/daemons", ct));
+
+                if (daemonsResp.StatusCode == HttpStatusCode.Unauthorized)
+                    return BuildToolResult(id, await AuthRejectionNotice.ForPersistentUnauthorizedAsync(apiRoot), isError: true);
+
+                ReviewerVendorsResult result;
+                if (!daemonsResp.IsSuccessStatusCode) {
+                    result = ReviewerVendorLookup.Aggregate(null, repoRoot, MachineId.Get(), driverVendor);
+                } else {
+                    var daemonsBody = await daemonsResp.Content.ReadAsStringAsync();
```
[the remark this answers is not in the document]
Comment (realtonyyoung) [reply src/Capacitor.Cli/Commands/McpFlowsServer.cs:420]: Fixed in d08fd62. The tool now reads the machine id via MachineId.ReadPersisted() instead of Get(), so it never persists machine.json and the read-only/side-effect-free contract holds. When no machine.json exists yet the id is null, which simply drops the same-machine filter for that one call.

File: src/Capacitor.Cli/Commands/ReviewerVendorLookup.cs [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    public static ReviewerVendorsResult Aggregate(
+            IReadOnlyList<DaemonVendorRecord>? daemons,
+            string? repoRoot,
+            string? requesterMachineId,
+            string? driverVendor,
+            bool schemaSkew = false,
+            int skippedRecords = 0) {
+        var resolved = !string.IsNullOrEmpty(repoRoot);
+        var repo     = new RepoIdent(repoRoot, resolved);
+
```
[the remark this answers is not in the document]
Comment (realtonyyoung) [reply src/Capacitor.Cli/Commands/ReviewerVendorLookup.cs:74]: Fixed in d08fd62. repo.identity now carries a non-sensitive owner/repo id (derived from repoInfo), never the absolute local repoRoot. The local path stays internal and is used only for the on-disk hosting match, so no local path leaks to the model. Added a unit test pinning that identity is not the local path.

File: src/Capacitor.Cli/Commands/ReviewerVendorLookup.cs [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+            return Empty(repo, driverVendor, daemons?.Count ?? 0, 0, skippedRecords, Reason.RepoUnresolved);
+        if (schemaSkew)
+            return Empty(repo, driverVendor, daemons?.Count ?? 0, 0, skippedRecords, Reason.SchemaSkew);
+        if (daemons is null)
+            return Empty(repo, driverVendor, 0, 0, skippedRecords, Reason.LookupFailed);
+
+        var norm = Normalize(repoRoot!);
+        var hosting = daemons
+            .Where(d => d.RepoPaths.Any(p => Normalize(p) == norm) &&
+                        (requesterMachineId is null || d.MachineId == requesterMachineId))
```
[the remark this answers is not in the document]
Comment (realtonyyoung) [reply src/Capacitor.Cli/Commands/ReviewerVendorLookup.cs:85]: Fixed in d08fd62. Path matching now unifies separators and folds case per-OS (OrdinalIgnoreCase on Windows and macOS, Ordinal on Linux), so a backslash-vs-slash or trailing-separator difference no longer yields a false no_repo_hosting_daemon. Added a separator-normalization test.

File: src/Capacitor.Cli/Commands/ReviewerVendorLookup.cs [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    /// mistaken for an authoritative empty set; a single malformed element is skipped and counted.
+    /// Only called on a 2xx; a non-2xx maps to <c>Aggregate(daemons: null, …)</c> = lookup_failed.</summary>
+    public static (IReadOnlyList<DaemonVendorRecord> Records, int Skipped, bool SchemaSkew) ParseDaemons(string body) {
+        JsonDocument doc;
+        try { doc = JsonDocument.Parse(body); } catch { return ([], 0, true); }
+
+        using (doc) {
+            if (doc.RootElement.ValueKind != JsonValueKind.Array)
+                return ([], 0, true);
+
```
[the remark this answers is not in the document]
Comment (realtonyyoung) [reply src/Capacitor.Cli/Commands/ReviewerVendorLookup.cs]: Fixed in d08fd62. ParseDaemons now uses the JsonElementExtensions helpers (IsArray / IsObject / IsString / Str / Arr / Bool) instead of raw JsonValueKind comparisons, per the repo JSON-handling standard.
